handler/src/main/java/io/netty/handler/ssl/OpenSslKeyMaterialManager.java

```diff
@@ -84,9 +85,46 @@ void setKeyMaterial(ReferenceCountedOpenSslEngine engine) throws SSLException {
         }
     }
 
-    void setKeyMaterial(ReferenceCountedOpenSslEngine engine, String[] keyTypes,
-                        X500Principal[] issuer) throws SSLException {
-        setKeyMaterial(engine.sslPointer(), chooseClientAlias(engine, keyTypes, issuer));
+    CertificateRequestedCallback.KeyMaterial keyMaterial(ReferenceCountedOpenSslEngine engine, String[] keyTypes,
+                                                         X500Principal[] issuer) throws SSLException {
+        String alias = chooseClientAlias(engine, keyTypes, issuer);
+        long keyBio = 0;
+        long keyCertChainBio = 0;
+        long pkey = 0;
+        long certChain = 0;
+
+        try {
+            // TODO: Should we cache these and so not need to do a memory copy all the time ?
+            PrivateKey key = keyManager.getPrivateKey(alias);
+            X509Certificate[] certificates = keyManager.getCertificateChain(alias);
+
+            if (certificates != null && certificates.length != 0) {
```

Comment:
nit: consider refactoring (less chance of branching and less indentation in code):

```java
if (certificates == null || certificates.length == 0) {
  return null;
}
...
```